parser.go

```diff
@@ -1,29 +1,111 @@
 package main
 
 import (
-	"strings"
-	"log"
+	"encoding/json"
+	"fmt"
 	req "github.com/imroc/req"
+	"io/ioutil"
+	"os"
+	"strings"
 )
 
-func nemoParser(pesan string, Sessions Session) (string, error){
-	for indexURL := 0; indexURL < strings.Count(pesan, "{{"); indexURL++ {
-		url := between(pesan, "{{", "}}")
+var (
+	// TODO use struct for key priority to allow setting config via file
+	// that might need extra config validation though
+	defSupportedRespKeys = map[string]int{
+		"message": 1,
+	}
+	defSupportedRespKeysConfig = "SUPPORTED_KEYS_CONFIG"
+
+	// TODO handle error differently
+	// these are only for visualization purpose
+	errReqErr           = func(dst string) string { return fmt.Sprintf("[req error: %s]", dst) }
+	errRespNotSupported = func(dst string) string { return fmt.Sprintf("[resp not supported error: %s]", dst) }
+)
 
+// nemoParser parses pesan for URL in {{url}} format
+// if no URL is found in pesan, pesan is returned as is
+// if URL is found, try POST request to url
+// currently only supports JSON response with message key
+func nemoParser(pesan string, Sessions Session) (string, error) {
+	urlCount := strings.Count(pesan, "{{")
+	if urlCount == 0 {
+		return pesan, nil
+	}
+	for i := 0; i < urlCount; i++ {
+		url := between(pesan, "{{", "}}")
 		r, err := req.Post(url, req.BodyJSON(Sessions))
 		if err != nil {
-			return "error occured, try again", err 
+			pesan = strings.Replace(pesan, fmt.Sprintf("{{%s}}", url), errReqErr(url), -1)
 		}
 
-		var pF pesanFetch
+		var m map[string]interface{}
+		r.ToJSON(&m)
 
-		r.ToJSON(&pF)
+		// TODO maybe calls this in main to setup
+		sk := supportKey(os.Getenv(defSupportedRespKeysConfig))
+		k := lookupKey(m, sk)
+		if k == "" {
+			pesan = strings.Replace(pesan, fmt.Sprintf("{{%s}}", url), errRespNotSupported(url), -1)
+			continue
+		}
+		if m[k] != "" {
+			pesan = strings.Replace(pesan, fmt.Sprintf("{{%s}}", url), fmt.Sprintf("%v", m[k]), -1)
+		}
+	}
 
-		log.Printf("%+v", pF)
+	return pesan, nil
+}
 
-		pesan = strings.Replace(pesan, "{{" + url + "}}", pF.Message, -1)
-		log.Println(pesan)
+// lookupKey looks up if key exists in a map based on priority
+// return empty string if key is not supported
+// currently only supports first level key
+// might want to improve the algorithm
+// i.e return once a supported key is found with the assumption
+// that response body has commonly used keys to denote response message
+// it also does not care if the config sets multiple keys with the same
+// priority, it uses whichever is found first
+func lookupKey(m map[string]interface{}, sup map[string]int) string {
```

Comment:
if you don't care about priority/handling multiple possible keys, change supported keys to string slice (or map[string]struct{} if you need to squeeze out performance)